mr/src/main/java/org/opensearch/hadoop/rest/commonshttp/CommonsHttpTransport.java

```diff
@@ -683,7 +758,12 @@ public Response execute(Request request) throws IOException {
 
         // when tracing, log everything
         if (log.isTraceEnabled()) {
-            log.trace(String.format("Tx %s[%s]@[%s][%s]?[%s] w/ payload [%s]", proxyInfo, request.method().name(), httpInfo, request.path(), request.params(), request.body()));
+            log.trace(String.format("Tx %s[%s]@[%s][%s]?[%s] w/ payload [%s]", proxyInfo, request.method().name(),
+                    httpInfo, request.path(), request.params(), request.body()));
+        }
+
+        if (settings.getAwsSigV4Enabled()) {
+            awsSigV4SignRequest(request, http, bodyBytes);
```

Comment:
Also add a test in `CommonsHttpTransportTests`?

Reply:
Added a test for the SigV4 signer.